### PR TITLE
refactor(shortcuts): extract matchShortcut helper for layout/macOS test coverage

### LINKED ISSUE #633: Extract matchShortcut helper from App.svelte for layout/macOS test coverage
## Context

Follow-up from PR #626 (keyboard shortcuts). The original plan-out included extracting a pure `matchShortcut(e: KeyboardEvent): ShortcutId | null` helper to `src/client/hooks/useAppShortcuts.ts`, but pragmatic execution chose in-place \`e.code\` conversion instead (lower blast radius across 15 context-dependent handlers).

The Critical macOS bug is fixed — Option+letter shortcuts now match via \`e.code\` so the alt-character produced by Mac's Western keyboard (\`†\`, \`µ\`, \`¬\`, \`å\`) no longer breaks them. But the test coverage that proves it is verbal-only: the diff is reviewable, but no unit test dispatches a \`KeyboardEvent({ code: "KeyT", key: "†", ctrlKey: true, altKey: true })\` and asserts the handler fires.

## Why this is worth doing eventually

1. **Mac CI confidence.** Bryan develops on Windows. Without a unit test, the only way a macOS regression surfaces is a real user reporting it.
2. **Layout-independence regression guard.** Today's \`e.code === "KeyT"\` pattern can be accidentally reverted to \`e.key === "t"\` in a future refactor and pass all existing Windows-only tests.
3. **Documentability.** A test file is the cleanest place to enumerate the 15 shortcuts and their modifier rules.

## Proposed shape (from the plan-out review)

Per the pr-test-analyzer's recommendation:

- New file: \`src/client/hooks/useAppShortcuts.ts\` exporting \`matchShortcut(e): ShortcutId | null\` and a \`ShortcutId\` union type.
- App.svelte's keydown handler becomes: \`const id = matchShortcut(e); if (id) { e.preventDefault(); dispatch[id](); }\`. The dispatch table holds the inline logic (editor focus, modal state, findState) so the helper stays pure.
- New file: \`tests/client/useAppShortcuts.test.ts\` with cases:
  - macOS Alt-letter: \`{ code: "KeyT", key: "†", ctrlKey: true, altKey: true }\` → \`"reopen-closed-tab"\`
  - macOS Alt-bracket: \`{ code: "BracketLeft", key: "“", altKey: true }\` → \`"prev-annotation"\`
  - Dvorak \`Ctrl+W\`: \`{ code: "KeyW", key: ",", ctrlKey: true }\` → \`"close-tab"\`
  - **Negative regression guard**: bare \`{ key: "†" }\` (no modifiers) → \`null\`
  - \`Ctrl+M\` vs \`Ctrl+Shift+M\` discrimination

## Risks

Some shortcuts are inherently context-dependent (\`Ctrl+F\` branches on outline visibility, \`Ctrl+G\` on findState presence). These can't cleanly map to a single \`ShortcutId\`. The helper should return either:
- A discriminated union (\`{ kind: "find", context: "outline" | "findbar" }\`), or
- An \`ID + context\` tuple where the dispatch table reads context.

Either is fine; pick the one that keeps App.svelte's keydown handler readable.

## Acceptance

- All 15 shortcuts dispatched via the helper, none retained as ad-hoc branches in App.svelte.
- Unit test file covers macOS Option-letter, Dvorak letter, negative case, and Ctrl+M / Ctrl+Shift+M discrimination.
- No behavioral regressions (verified via existing E2E suite).

## Origin

PR #626 review pass; plan-out plan section "PR #626 New test files". Pragmatically deferred during execution; this issue captures the deferred work.

### PR DESCRIPTION
**Pull request by bloknayrb**
## Summary
- Extract the pure keyboard-event-to-shortcut matching logic from `App.svelte`'s inline keydown handler into `src/client/hooks/useAppShortcuts.ts`, exposing a `ShortcutId` union and a `matchShortcut(e): { id, context } | null` helper that's testable in isolation.
- App.svelte now mounts a small `Partial<Record<ShortcutId, ShortcutHandler>>` dispatch table; each entry keeps its idiosyncratic side-effect logic (preventDefault policy, editor commands, modal state, findState, notification toasts) so the helper stays pure.
- Matcher gating is faithful to the legacy else-if/outer-if semantics — branches the old code did NOT gate on alt/shift (Ctrl+S, Ctrl+N, Ctrl+W, Ctrl+O, Ctrl+1..9, Ctrl+Alt+T, Ctrl+Alt+M, Ctrl+Alt+A, …) remain ungated, so no existing edge-case behavior changes.
- New test file `tests/client/useAppShortcuts.test.ts` covers all required cases (macOS Option-letter via `e.code`, macOS Alt-bracket, Dvorak Ctrl+W, bare `†` negative regression guard, Ctrl+M vs Ctrl+Shift+M discrimination) plus context payloads, settings/select-all gating priorities, and explicit no-modifier-gate preservation for the legacy chain (50 tests, all passing).

## Test plan
- [x] `npm run typecheck` — clean (0 errors, 0 warnings across 841 files)
- [x] `npx vitest run tests/client/useAppShortcuts.test.ts` — 50/50 passing
- [x] `npx vitest run tests/client/` — 734/734 passing
- [ ] `npm run test:e2e -- shortcuts` — skipped (worktree missing `tsx` in node_modules; playwright `webServer` fails to start the dev server)
- [ ] Manual: verify Ctrl+A / Ctrl+S / Ctrl+W / Ctrl+O / Ctrl+1..9 / Ctrl+,/Ctrl+Shift+, / Ctrl+Shift+P / Ctrl+N / Ctrl+/ / "?" / Ctrl+Shift+M / Ctrl+Alt+T / Ctrl+F / Ctrl+Shift+F / Ctrl+G / Ctrl+Shift+G / Ctrl+Enter / Ctrl+Shift+Enter / Ctrl+Alt+M / Ctrl+Alt+A / Alt+Shift+Arrow / Alt+] / Alt+[ / Alt+L all still fire in the desktop / browser app.

Fixes #633

https://claude.ai/code/session_01XXe5nk1E68VgrQ55yunVe9

---
_Generated by [Claude Code](https://claude.ai/code/session_01XXe5nk1E68VgrQ55yunVe9)_

### CODE REVIEW

File: src/client/App.svelte [hunk 1]
```diff
@@ -682,244 +682,230 @@ function captureSelectionForChat() {
...
-      } else if (e.shiftKey && !e.altKey && e.code === "KeyM") {
-        if (shouldIgnoreShortcut(e)) return;
-        e.preventDefault();
-        modeState.setTandemMode(modeState.tandemMode === "solo" ? "tandem" : "solo");
-      } else if (e.altKey && e.code === "KeyT") {
-        if (shouldIgnoreShortcut(e)) return;
-        e.preventDefault();
-        void reopenClosedTab();
-      }
+    e.preventDefault();
```
Comment (bloknayrb) [inline src/client/App.svelte:694]: Behavior drift vs master for bare `?`. The old outermost handler toggled `showHelp` **without** calling `e.preventDefault()`:

```js
if (e.key === "?") {
  const el = e.target as HTMLElement;
  if (el.tagName === "INPUT" || el.tagName === "TEXTAREA" || el.isContentEditable) return;
  showHelp = untrack(() => !showHelp);   // no preventDefault
  return;
}
```

The unified `toggle-help` handler now always calls `e.preventDefault()`, which applies to bare `?` too (it's reached via the matcher's top-level `e.key === "?"` branch as well as Ctrl+/). In practice bare `?` outside an input/contenteditable has no useful default behavior to suppress, so impact is near-zero — but this is the one spot where the refactor isn't strictly faithful to the legacy semantics the PR claims to preserve. Flagging since the PR's stated goal is "no existing edge-case behavior changes." (Ctrl+/ *did* preventDefault originally, so that path is fine.)

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/client/hooks/useAppShortcuts.ts [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+ *
+ * Note: this helper does NOT apply input-field suppression
+ * (`shouldIgnoreShortcut`) or the "?" `INPUT/TEXTAREA/contenteditable`
+ * guard — those depend on `e.target`, which the dispatch table handles per
+ * shortcut. Centralizing the suppression here would over-suppress
+ * shortcuts that intentionally fire in form fields (Ctrl+S, Ctrl+F, etc.).
+ */
+export function matchShortcut(e: KeyboardEventLike): ShortcutMatch | null {
+  // IME composition: never dispatch a shortcut mid-composition.
+  if (e.isComposing) return null;
```
Comment (bloknayrb) [inline src/client/hooks/useAppShortcuts.ts:105]: Behavior drift vs master: this top-of-matcher `isComposing` guard now blanket-suppresses **every** App-level shortcut during IME composition. In the legacy code, only the branches that called `shouldIgnoreShortcut(e)` and the two settings predicates (`isSettingsShortcut`/`isSettingsModalShortcut`) checked `isComposing`. Branches like Ctrl+S (save), Ctrl+F (find), Ctrl+Shift+P (palette), Ctrl+Alt+M (comment-on-selection), Ctrl+Alt+A (authorship), and bare `?`/Ctrl+/ (help) had **no** IME check and would still fire mid-composition.

So Ctrl+S / Ctrl+F pressed while an IME composition is active is now a no-op where master would have saved / opened find. This is plausibly an intentional and beneficial hardening (and most browsers don't report Ctrl-combos with `isComposing === true` anyway), but it's not strictly faithful to master. Worth a one-line note in the PR description or a comment confirming the broadened IME gate is deliberate.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/client/hooks/useAppShortcuts.ts [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+ *    AZERTY users hit the right shortcut and macOS Option-letter chords don't
+ *    miss because Option produces alt chars like "†"/"µ".
+ *  - Digit shortcuts also match on `e.code` ("Digit1"…).
+ *  - Bracket shortcuts match on `e.code` ("BracketLeft"/"BracketRight") so the
+ *    macOS Alt-bracket chord (Alt+[ → '"', Alt+] → '"') doesn't miss.
+ *  - Arrow / Enter / Comma / Slash / Question shortcuts that have no
+ *    layout-variation use `e.key` (or the appropriate `e.code`).
+ *
+ * Context fields:
+ *  - `findShift`/`findGShift` — whether Shift was held, so the dispatch can
```
Comment (bloknayrb) [inline src/client/hooks/useAppShortcuts.ts]: Stale doc comment: this "Context fields" block references field names that don't exist in the implemented `ShortcutContext` — it lists `findShift`/`findGShift`/`acceptDismissShift`, but the actual interface (below) collapses all three into a single `shift?: boolean`. Update the comment to match (`shift` + `tabIndex`) so future readers don't go looking for fields that aren't there. Doc-only, no runtime impact.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by bloknayrb — COMMENTED** [review]
Code review (xhigh effort, recall mode) of the `matchShortcut` extraction.

**Overall: the refactor is faithful.** I traced every legacy branch (inner ctrl/meta else-if chain + the standalone outer `if` blocks) against the new matcher, modifier combination by modifier combination — including all the no-gate edge cases the worker self-caught (Ctrl+Alt+S→save, Ctrl+Alt+W→close-tab, Ctrl+Shift+Alt+M→comment, Ctrl+Alt+Enter→accept, etc.). Routing matches master for every combination I checked. Dispatch-table coverage is complete (all 22 `ShortcutId`s have entries) and context passing (`tabIndex` for pick-tab, `shift` for find/find-nav/accept-dismiss) is consistent between matcher and dispatch. No correctness bugs found.

The 3 findings below are all **low-severity behavior drifts / doc nits**, not breakages:

1. `App.svelte` bare `?` now calls `preventDefault()` (master did not).
2. `useAppShortcuts.ts` IME `isComposing` guard now applies to *all* shortcuts (master only gated a subset).
3. `useAppShortcuts.ts` stale `ShortcutContext` doc-comment field names.

Severity breakdown: 0 critical / 0 high / 0 medium / 3 low. 50/50 new unit tests pass locally.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/client/hooks/useAppShortcuts.ts [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+    // Ctrl+Shift+, → new SettingsModal. Tested BEFORE Ctrl+, so a future edit
+    // to the popover predicate can't accidentally claim the shifted form.
+    // Legacy `isSettingsModalShortcut`: `code === "Comma" && shiftKey`.
+    if (e.shiftKey && e.code === "Comma") {
+      return { id: "settings-modal" };
+    }
+    // Ctrl+, → settings popover. Legacy `isSettingsShortcut`: rejects shift.
+    if (!e.shiftKey && e.code === "Comma") {
+      return { id: "settings" };
+    }
```
Comment (bloknayrb) [inline src/client/hooks/useAppShortcuts.ts:135]: **Low — dual source of truth for the Comma settings shortcut.** This re-implements the gating that already lives in `isSettingsModalShortcut`/`isSettingsShortcut` (`hooks/useSettingsShortcut.ts`). After this PR, `App.svelte` no longer imports those predicates, so `useSettingsShortcut.ts` is now dead production code — only `tests/client/useSettingsShortcut.test.ts` references it. The matcher's behavior here is correct and faithful (legacy required `ctrl||meta` + shift discrimination, both preserved by the enclosing `if (mod)` block), but there are now two definitions of the same shortcut that can drift independently.

Consider either (a) calling the existing predicates here so there's one definition, or (b) deleting `useSettingsShortcut.ts` + its test now that the matcher owns the logic. Leaving both in place is the worst of the two — the old predicate looks live but isn't.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/client/hooks/useAppShortcuts.ts [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+  const mod = e.ctrlKey || e.metaKey;
+
+  // "?" → toggle help. Legacy outermost handler had NO modifier gate; preserve
+  // that so Ctrl+Shift+/ (which produces e.key === "?" on US layouts) still
+  // routes to help. The Option+t-produces-"†" negative regression guard works
+  // because e.key === "†" doesn't match "?" — the matcher reads e.key, not a
+  // character-class.
+  if (e.key === "?") {
+    return { id: "toggle-help" };
+  }
```
Comment (bloknayrb) [inline src/client/hooks/useAppShortcuts.ts:116]: **Low — test coverage gap for the documented `Ctrl+Shift+/` → `?` path.** The doc comment (lines 109-113) explicitly promises that `Ctrl+Shift+/`, which produces `e.key === "?"` on US layouts, still routes to `toggle-help` via this top-of-function branch — a deliberate fidelity guarantee vs. the legacy outermost handler. But the only `?` test in `useAppShortcuts.test.ts:190` is bare `?` with no modifiers. There's no regression guard that `evt({ key: "?", ctrlKey: true, shiftKey: true })` returns `{ id: "toggle-help" }`, so a future edit that wraps this branch in a `!mod` guard would silently break the documented behavior. One assertion would lock it in.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/client/hooks/useAppShortcuts.ts [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+    }
+
+    // Alt+] / Alt+[ → next / previous annotation. No Shift.
+    if (!e.shiftKey) {
+      if (e.code === "BracketRight") return { id: "annotation-next" };
+      if (e.code === "BracketLeft") return { id: "annotation-prev" };
+    }
+
+    // Alt+L → select containing block.
+    if (!e.shiftKey && e.code === "KeyL") return { id: "select-block" };
```
Comment (bloknayrb) [inline src/client/hooks/useAppShortcuts.ts:242]: **Low — untested `!shiftKey` rejection on the alt-only bracket / `KeyL` branches.** Lines 236-242 deliberately require `!e.shiftKey` so that `Alt+Shift+[`, `Alt+Shift+]`, and `Alt+Shift+L` fall through to `null` (and the shift+arrow panel toggles own the shifted alt space). The suite tests the positive bracket/`KeyL` matches and the `Alt+ArrowLeft`-without-shift rejection, but never asserts that `Alt+Shift+]` / `Alt+Shift+[` / `Alt+Shift+L` return `null`. If a later edit drops `!e.shiftKey` here, the matcher would start claiming those chords and nothing in the 50-test suite would fail. Three `toBeNull()` cases would close the gap and document the intent.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/client/hooks/useAppShortcuts.ts [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+ *
+ * Note: this helper does NOT apply input-field suppression
+ * (`shouldIgnoreShortcut`) or the "?" `INPUT/TEXTAREA/contenteditable`
+ * guard — those depend on `e.target`, which the dispatch table handles per
+ * shortcut. Centralizing the suppression here would over-suppress
+ * shortcuts that intentionally fire in form fields (Ctrl+S, Ctrl+F, etc.).
+ */
+export function matchShortcut(e: KeyboardEventLike): ShortcutMatch | null {
+  // IME composition: never dispatch a shortcut mid-composition.
+  if (e.isComposing) return null;
```
Comment (bloknayrb) [inline src/client/hooks/useAppShortcuts.ts:105]: **Info — confirming the two intentional behavior drifts (both acceptable, flagging for the record).**

1. **`isComposing` broadening.** The legacy handler had no top-level IME guard; it only filtered composition inside the branches that called `shouldIgnoreShortcut` (Ctrl+W/O, Ctrl+1-9, Ctrl+Shift+M, Ctrl+Alt+T, Ctrl+G, Alt+Shift+Arrow, Alt+Bracket, Ctrl+Enter, Alt+L). Branches that did *not* call it — Ctrl+A, Ctrl+S, settings, Ctrl+Shift+P, Ctrl+N, Ctrl+/, Ctrl+F, Ctrl+Alt+M, Ctrl+Alt+A, bare `?` — used to fire mid-composition. This `if (e.isComposing) return null` now suppresses all of them while an IME is active. That's a strict improvement (no shortcut should fire mid-composition) and matches the helper's documented contract, so no change requested — just noting it diverges from a literal reading of the legacy cascade.

2. **Bare `?` now triggers `preventDefault()`.** Legacy bare-`?` toggled help *without* `preventDefault()`; the `toggle-help` dispatch entry (App.svelte:707) now calls it. Harmless for a `?` keystroke outside form fields (the guard still lets it through in INPUT/TEXTAREA/contenteditable before preventDefault), and consistent with the Ctrl+/ path. Recorded for completeness.

Everything else I checked — branch ordering vs. the legacy else-if cascade, alt/shift gating on every combination (select-all vs toggle-authorship, toggle-mode vs comment-on-selection, Ctrl+Alt+Shift+M, the no-gate Ctrl+S/N/W/O/Digit/Ctrl+Alt+T), `tabIndex`/`shift` context consistency matcher↔dispatch, full 22-member `ShortcutId` dispatch coverage, and the keydown `$effect` (still zero reactive deps — reads only the non-reactive `dispatch` const + imported `matchShortcut`, so it mounts once exactly like the legacy effect; handlers read live reactive state lazily at call time, no staleness) — is faithful. Nice refactor.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by bloknayrb — COMMENTED** [review]
## Svelte-5 + behavior-fidelity review — PR #820 (`matchShortcut` extraction)

Reviewed head `9e4bcde` against `master` (base `b9d37d5`). The refactor is faithful to the legacy `App.svelte` else-if/outer-if keydown cascade and the dispatch table is complete. No correctness or reactivity defects. 3 Low (maintainability + test-coverage) and 2 Info (intentional, documented drifts) findings posted inline.

**Fidelity verification (all faithful):**
- Branch ordering preserves the legacy cascade — earlier-branch shadowing matches (e.g. select-all's `!alt` gate routes Ctrl+Alt+A to toggle-authorship; toggle-mode's `!alt` routes Ctrl+Alt+M to comment; Ctrl+Alt+Shift+M → comment).
- Ungated legacy branches (Ctrl+S/N/W/O, Ctrl+1-9, Ctrl+Alt+T, Ctrl+F/G, Ctrl+Enter, Ctrl+Alt+M/A) remain ungated — no alt/shift over- or under-gating introduced.
- Comma settings gating (`shift` → modal, `!shift` → popover, both requiring ctrl/meta) matches `isSettingsModalShortcut`/`isSettingsShortcut`.
- `tabIndex` / `shift` context is produced by the matcher and consumed consistently in `pick-tab` / `find` / `find-nav` / `annotation-accept-or-dismiss`.
- All 22 `ShortcutId` members have a dispatch handler.
- The keydown `$effect` has zero reactive dependencies (reads only the non-reactive `dispatch` const + imported pure `matchShortcut`), so it mounts once — identical lifecycle to the legacy effect. Handlers read reactive state lazily at invocation, no staleness. No Svelte-5 reactivity regression.

**Intentional drifts (acceptable, flagged inline):** top-level `isComposing` guard now suppresses the few shortcuts that previously fired mid-IME-composition; bare `?` now calls `preventDefault()`.

**Tests:** 50/50 pass locally; related `useSettingsShortcut`/`useTabKeyboardShortcuts` suites (18) still green. Coverage is strong on macOS Option-letter / Alt-bracket / Dvorak / Ctrl+M-discrimination / no-gate preservation; gaps noted inline (Ctrl+Shift+/→? path; Alt+Shift bracket/`KeyL` rejection).

**Merge recommendation: approve / safe to merge** once the author decides on the `useSettingsShortcut.ts` dead-code question (reuse vs. delete). The two test-coverage gaps are nice-to-have hardening, not blockers.

---
_Generated by [Claude Code](https://claude.ai/code)_
